pkg/adaptor/cloud/vsphere/provider.go

```diff
@@ -35,6 +35,11 @@ func NewProvider(config *Config) (cloud.Provider, error) {
 
 	logger.Printf("vsphere config: %#v", config.Redact())
 
+	err := checkConfig(config)
+	if err != nil {
```

Comment:
Should the error be logged here ?

Reply:
All errors from NewProvider are logged in cmd/cloud-api-adaptor/main.go
func main() {

	starter, err := config.Setup()
	if err != nil {
		fmt.Fprintf(os.Stderr, "%s: %s\n", os.Args[0], err)
		cmd.Exit(1)
	}

So if I wanted to be consistent I should remove this logging in NewProvider and return the formatted error
govmomiClient, err := NewGovmomiClient(*config)
        if err != nil {
                logger.Printf("Error creating vcenter session for cloud provider:  %s", err)
                return nil, err
        }

Reply:
thanks for the clarification